src/cpp/rtps/writer/StatelessWriter.cpp

```diff
@@ -947,9 +947,13 @@ bool StatelessWriter::get_connections(
         //! intraprocess
         for_matched_readers(matched_local_readers_, [&connection, &connection_list](ReaderLocator& reader)
                 {
-                    connection.guid(fastdds::statistics::to_statistics_type(reader.local_reader()->getGuid()));
-                    connection.mode(fastdds::statistics::ConnectionMode::INTRAPROCESS);
-                    connection_list.push_back(connection);
+                    LocalReaderPointer::Instance local_reader = reader.local_reader();
+                    if (local_reader)
+                    {
+                        connection.guid(fastdds::statistics::to_statistics_type(local_reader->getGuid()));
+                        connection.mode(fastdds::statistics::ConnectionMode::INTRAPROCESS);
+                        connection_list.push_back(connection);
+                    }
```

Comment:
We could avoid getting the pointer to the local reader by using `ReaderLocator::remote_guid()` to retrieve the reader's GUID